Privacy: Display both date and time for older privacy requests

Sukhendu2002 opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Trac ticket: https://core.trac.wordpress.org/ticket/44267

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

Sukhendu2002 committed:
Show time alongside date for older privacy requests

Sukhendu2002 committed:
Fix PHPCS Errors

Sukhendu2002 committed:
Fix PHPCS Errors

SirLouen commented on src/wp-admin/includes/class-wp-privacy-requests-table.php:
```diff
@@ -487,7 +487,15 @@ protected function get_timestamp_as_date( $timestamp ) {
+
+		return sprintf(
+			/* translators: 1: Date, 2: Time */
+			__( '%1$s at %2$s' ),
```
Why have you chosen this format? In the report they were discussing multiple alternatives, and this was discarded.

joedolson replied:
Since there's been no follow up from the original author, I went ahead and updated this based on the ticket discussion.

joedolson committed:
Updates date/time format as discussed in ticket

Updates date/time formats and tests to use the last discussed format from ticket #44267. Sets formats to match the format used in the Comments table, and makes translatable so localization can customize as per their standard expectations.

joedolson committed:
Merge branch 'trunk' into pr/8709

joedolson committed:
Reduce font-weight of completed date

joedolson commented:
<img width="1159" height="205" alt="Screenshot 2025-09-28 at 5 46 23 PM" src="https://github.com/user-attachments/assets/127d2c35-f173-4c1a-9028-84b7c6dc3338" />

Screenshot showing updated date/time format.

joedolson committed:
Merge branch 'trunk' into pr/8709

joedolson committed:
Merge branch 'trunk' into pr/8709
